src/plugins/discover/public/application/main/services/discover_app_state_container.test.ts

```diff
@@ -161,4 +161,66 @@ describe('Test discover app state container', () => {
       );
     });
   });
+
+  describe('isEqualState', () => {
+    const initialState = {
+      index: 'the-index',
+      columns: ['the-column'],
+      sort: [],
+      query: { query: 'the-query', language: 'kuery' },
+      filters: [],
+      interval: 'auto',
+      hideChart: true,
+      sampleSize: 100,
+      viewMode: VIEW_MODE.DOCUMENT_LEVEL,
+      savedQuery: undefined,
+      hideAggregatedPreview: true,
+      rowHeight: 25,
+      headerRowHeight: 25,
+      grid: {},
+      breakdownField: 'the-breakdown-field',
+    };
+
+    test('returns true if the states are equal', () => {
+      expect(isEqualState(initialState, { ...initialState })).toBeTruthy();
+    });
+
+    test('handles the special filter change case correctly ', () => {
+      // this is some sort of legacy behavior, especially for the filter case
+      const previousState = { initialState, filters: [{ index: 'test', meta: {} }] };
+      const nextState = {
+        initialState,
+        filters: [{ index: 'test', meta: {}, $$hashKey: 'hi' }],
```

Comment:
@lukasolson I'm interested in the $$hashKey filter variable, we have to omit from `equal` detection. This is Angular, and it's no longer in use, but there could be filters out there, legacy one, that are using this, right?

Reply:
Yes, it's from Angular days. I could be wrong, but I don't think it ever made it into the URL, it was just part of the in-memory state, so I *think* it can be removed.

Reply:
Thx, we would need to double check for sure 👍